web/packages/core/src/shadow-template.ts

```diff
@@ -218,13 +222,91 @@ ruffleShadowTemplate.innerHTML = `
             border-bottom: 1px solid lightgray;
             margin: 2px;
         }
+
+        #preloader {
+            display: flex;
+            flex-direction: column;
+            background: var(--preloader-background, var(--ruffle-blue));
+            width: 60%;
+            height: auto;
+            align-items: center;
+            justify-content: center;
+            aspect-ratio: 2;
+            max-height: 100%;
+            position: relative;
+            top: 50%;
+            left: 50%;
+            transform: translate(-50%, -50%);
+        }
+
+        .loadbar {
+            width: 100%;
+            max-width: 316px;
+            max-height: 10px;
+            height: 20%;
+            margin-top: -3%;
+            background: #253559;
+        }
+
+        .loadbarInner {
+            width: 0px;
+            max-width: 100%;
+            height: 100%;
+            background: var(--ruffle-orange);
+        }
+            
+        .logo {
+            display: var(--logo-display, block);
+            max-width: 380px;
+            max-height: 150px;
+            resize: horizontal;
+            width: 100%;
+            height: 100%;
+        }
+
+        .loadingAnimation {
+            max-width: 28px;
+            max-height: 28px;
+            aspect-ratio: 1;
+            width: 10%;
+            padding-bottom: 5%;
+        }
+
+        .loading {
+            transform-origin: 50% 50% 0px;
+        }
+
+        .spinner {
+            stroke-dasharray: 180;
+            stroke-dashoffset: 0;
+            stroke: var(--ruffle-orange);
+            transform-origin: 50% 50%;
+            animation: rotate 1.5s linear infinite;
+        }
+
+        @keyframes rotate {
+            0% { stroke-dashoffset: 135; }
```

Comment:
If `stroke-dashoffset` doesn't change throughout the animation, why including it here? Please move it to `.spinner`. This makes the `0%` keyframe empty, so it can be removed. And also, the `100%` keyframe can be named `to` for simplicity.